### ISSUE TITLE
Conax Ca and Fixed/Static Constant Cw

### ISSUE THREAD

**Issue by thejackal2017** [neo7530/conax-ca#3]
Hi , After many tries i managed to make the ecmg and emmg to work with cesbo astra 

ECMG:
![ecmg](https://user-images.githubusercontent.com/31767937/208286705-e9fa39b4-8214-469f-af97-e0e921360c10.JPG)

EMMG:
![emmg](https://user-images.githubusercontent.com/31767937/208286713-25e2b6e5-77c0-4508-88f4-ca32323a0bb4.JPG)

is there anyway to make the cw keys static/Fixed/Unchanged for a while for example 1 hour for testing purposes ? 

because if i changed the crypto period from astra to 10 sec for example every 10 seconds a new request is sent with a new cw keys and i want a way to make the request still going in small intervals , but with the same cw keys .

I couldn't find the source anywhere to make such changes . 

So if this is possible please tell how can i make such scenario to work . 

Regards

**Comment by neo7530** [comment]
Hi.
The cw interval comes from the mux. The ecmg gets the actual used cw as plaintext, does the encryption and send the encryted cw back to the mux. The mux does  multiplex the ecm to ecm-pid. So just figure out if astra can use fixed cws...

**Comment by thejackal2017** [comment]
So are saying that the ecmg has nothing to to do with cw keys changing and all that it does is taking the cw keys  that are sent by astra and encrypt them ?

**Comment by neo7530** [comment]
Yes. This is the way it works. The ecmg does only encryption and structurizing the encryted cw, so the smartcard can understand and decrypt the cw. The dvbcsa-module ins astra or tsduck or whatever muxer is used, does the Video encryption and sends the used cws as plaintext to the ecmg. The emm part is done in emmg. The mux doesn't know anything about emm.

**Comment by thejackal2017** [comment]
thanks again for the replay , now this makes a lot of sense , but the main problem with astra is that the open source version of it is too old and back to almost 9 years i don't think cas was applicable that time and now they are working with paid and closed version so i think nothing to edit there . 

I have another question , let's imagine that the ecmg always return the same encrypted cw keys despite the real keys that astra sends to the ecmg , so what ever plain text keys astra sends , the ecmg return the same result to astra (the static keys from the point of view of ecmg) will this works in maintaing the cw keys from changing ? 

is such scenario possible ?

**Comment by neo7530** [comment]
This is why i'm using tsduck for muxing and scrambling. I don't have any experiences with astra, so in this point i can't help you anyway with this.

**Comment by thejackal2017** [comment]
@neo7530 , thanks a lot for uploading the source code , this was enough for solving the problem .

Is there any chance that you will work on other ecmg(s) in the future for any of the following Ca Systems ?
Biss 2 (mode CA)
Cryptoworks
Irdeto 2
Nagra 2
Videoguard
Viaccess v1, v2, v3

Regards

**Comment by neo7530** [comment]
Hi.

No, i dont mind on working with other Systems. I ve done one for irdeto 1 and the conax like Thing. This was a proof of concept only and some Code to learn programming in c.

Kind regards
Neo.
